src/perl/Permabit/VolumeGroup.pm

```diff
@@ -177,27 +172,58 @@ sub createThinVolume {
     $ksize = $kfree;
   }
 
-  # Create a thin volume in an existing volume group. This is a two step
-  # process where we create a thin pool first, then a thin volume.
-  # If lvcreate asks for a yes/no on whether to wipeout a filesystem
-  # signature (it does on RHEL7), the input redirection will cause the
-  # answer to be "no". The volume is not immediately enabled.
-  $machine->runSystemCmd("sudo lvcreate --name $name-pool --type=thin-pool"
-                         . " --extents 100\%FREE --yes $config"
+  # The first time a thin pool is created, lvm will create both a metadata device
+  # and a spare metadata LV in the VG. The spare device behavior can be controlled with
+  # the option --poolmetadataspare y|n. Unfortunately, if we try to create a pool using
+  # --poolmetadataspare n, the output will still generate a WARNING to stderr, thus
+  # causing our perl code to fail.
+  #
+  # As such, we have to take in account the size of both the metadata device and the
+  # spare metadata device when calculating how much space the pool takes as it doesn't
+  # take that into account itself. So we take the overall size asked and add the size
+  # needed for both the metadata and spare metadata devices. Then we see if the VG
+  # has sufficient space. Currently that space is 30 extents for each device.
+  my $metaDataSize = 60 * ($self->{physicalExtentSize} / $KB);
+  my $fullSize = $ksize + $metaDataSize;
```

Comment:
The comment makes this much clearer now.
(60 * $self->{physicalExtentSize}) / $KB as a spelling communicate the idea of "60 extents" better, to me, but it's a minor thing.

Reply:
I've changed this to do this ordering